Core/JsRuntime/Source/JsRuntime.cpp

```diff
@@ -0,0 +1,61 @@
+#include "JsRuntime.h"
+
+namespace Babylon
+{
+    namespace
+    {
+        static constexpr auto JS_RUNTIME_NAME = "runtime";
+        static constexpr auto JS_WINDOW_NAME = "window";
+    }
+
+    JsRuntime::JsRuntime(DispatchFunctionT&& dispatchFunction)
+        : m_dispatchFunction{ dispatchFunction }
+    {
+    }
+
+    void JsRuntime::Initialize(Napi::Env env, DispatchFunctionT dispatchFunction)
+    {
+        JsRuntime* runtime = new JsRuntime(std::move(dispatchFunction));
+        runtime->AddJavaScriptReference(env, true);
```

Comment:
I think it's clearer if you passed a lambda instead of true here:

```suggestion
        runtime->AddJavaScriptReference(env, [](Napi::Env, JsRuntime* jsRuntime) { delete jsRuntime; });
```

Reply:
Since this is a protected function, that externalizes a little bit of implementation knowledge from this class that I'm not sure needs to be externalized. At present, the fact that `JsRuntime` is stored up on the JavaScript as a `Napi::External` which may or may not have a custom deleter is an implementation detail; inheritors need to know whether or not JavaScript owns the JsRuntime, but they don't need to know why that matters or what's different as a result. We've only got one inheritor at present, and it's just the inheritance contract anyway, so I don't think it's too big of a deal either way; but unless you feel strongly otherwise, my _slight_ preference would be to leave this argument as a boolean to keep the fact that this is stored as a `Napi::External` out of the contract. Again, though, I don't feel _too_ strongly about it either way.